### ISSUE TITLE
include: using options "separate" and "attributes" together throws TypeError: Cannot read property 'push' of undefined

### ISSUE THREAD

**Issue by wolframkleis** [sequelize/sequelize#8228]
## What you are doing?

I try to include hasMany association with "include" that uses both "separate" option and "attributes" option

```js
const Parent = sequelize.define('Parent',
  {	
    id:  		{ type: Sequelize.INTEGER, primaryKey: true},
    name:  		{ type: Sequelize.STRING(50)}
  });

const Child = sequelize.define('Child',
  {	
    id:  		{ type: Sequelize.INTEGER, primaryKey: true},
	name:  		{ type: Sequelize.STRING(50)}
  });

Parent.hasMany(Child,{as: 'children'});

sequelize.sync({force:true,logging: false})
.then(()=>Parent.create({
    id: 1, 
	name: 'Parent1',
	children: [{id: 1,name: 'Child1'},{id: 2,name: 'Child2'}]
  },{include: [ {association: 'children'} ]}))

//first, we use an include with only  "attributes"  option:  works fine
.then(()=>Parent.findAll({
  include: {
	association: 'children',
	attributes: ['name']
  }}))

//next we use an include with only "separate" option: works fine, too
.then(()=>Parent.findAll({
  include: {
	association: 'children',
	separate: true
  }}))

//finally, we try the include with both "separate" and "attributes"  options:  FAILS
.then(()=>Parent.findAll({
  include: {
	association: 'children',
	attributes: ['name'],
	separate: true
  }}))
```
## What do you expect to happen?
I expect that you can use both options together.
It should be possible to select certain attributes only AND to request separate queries (to avoid the 63 character issue when using PostgreSQL, for example) 

## What is actually happening?
when using both options together, findAll fails with this error:
```
Unhandled rejection TypeError: Cannot read property 'push' of undefined
    at Model.findAll.then.results (C:\test\constraints\node_modules\Sequelize\lib\associations\has-many.js:230:66)
    at tryCatcher (C:\test\constraints\node_modules\bluebird\js\release\util.js:16:23)
    at Promise._settlePromiseFromHandler (C:\test\constraints\node_modules\bluebird\js\release\promise.js:512:31)
    at Promise._settlePromise (C:\test\constraints\node_modules\bluebird\js\release\promise.js:569:18)
    at Promise._settlePromise0 (C:\test\constraints\node_modules\bluebird\js\release\promise.js:614:10)
    at Promise._settlePromises (C:\test\constraints\node_modules\bluebird\js\release\promise.js:693:18)
    at Async._drainQueue (C:\test\constraints\node_modules\bluebird\js\release\async.js:133:16)
    at Async._drainQueues (C:\test\constraints\node_modules\bluebird\js\release\async.js:143:10)
    at Immediate.Async.drainQueues (C:\test\constraints\node_modules\bluebird\js\release\async.js:17:14)
    at runCallback (timers.js:651:20)
    at tryOnImmediate (timers.js:624:5)
    at processImmediate [as _immediateCallback] (timers.js:596:5)
```
__Dialect:__ postgres 
__Database version:__ 9.4
__Sequelize version:__ 4.7.5

**Comment by wolframkleis** [comment]
not stale

**Comment by antonioasmoura** [comment]
Facing the same problem!

**Comment by kgiannelos** [comment]
Same here, facing the same.

**Comment by sushantdhiman** [comment]
Looks same as https://github.com/sequelize/sequelize/issues/7514, which has been fixed https://github.com/sequelize/sequelize/pull/9396 and available in `5.0.0-beta.5`

**Comment by kgiannelos** [comment]
It looks the same indeed, but it could be different as well. In my case, adding the foreign keys to the model definition and also in the the attributes list in the include object did not help. However what worked out in my case was to remove completely the attributes list form the include. This worked even without declaring the foreign keys explicitly in the model definition.
